pyQuARC/code/datetime_validator.py

```diff
@@ -73,3 +75,67 @@ def compare(first, second, relation):
             "valid": result,
             "value": (str(first), str(second))
         }
+
+    @staticmethod
+    def validate_datetime_against_granules(datetime, collection_shortname, sort_key, time_key):
+        """
+        Validates the collection datetime against the datetime of the last granule in the collection
+
+        Args:
+            datetime (str): datetime string
+            collection_shortname (str): ShortName of the parent collection
+            sort_key (str): choice of start_date and end_date
+            time_key (str): choice of time_end and time_start
+        Returns:
+            (dict) An object with the validity of the check and the instance
+        """
+        granules = requests.get(f'{CMR_URL}/search/granules.json?short_name={collection_shortname}&sort_key[]=-{sort_key}').json()
+
+        if len(granules['feed']['entry']) > 0:
```

Comment:
Check with the ARC team if we need to select random or last granule.

Reply:
The check says the 'oldest': https://github.com/NASA-IMPACT/pyQuARC/issues/138